### PR TITLE
feat(taskbroker): Add the taskbroker to devservices

### PR DESCRIPTION
**Pull request by evanh**
Add the ability to start a local taskbroker instance, behind a flag. This image is updated every
time a PR is merged into taskbroker main.

Also added two small scripts to help with local testing.

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -2412,6 +2412,20 @@ def custom_parameter_sort(parameter: dict) -> tuple[str, int]:
             "platform": "linux/amd64",
         }
     ),
+    "taskbroker": lambda settings, options: (
+        {
+            "image": "ghcr.io/getsentry/taskbroker:latest",
+            "ports": {"50051/tcp": 50051},
+            "environment": {
+                "TASKBROKER_KAFKA_CLUSTER": "sentry_kafka:9093",
```
Comment (markstory) [inline src/sentry/conf/server.py:2420]: I think we'll need a different host when `USE_NEW_DEVSERVICES` is active. We use

```python
host = "kafka-kafka-1" if os.environ.get("USE_NEW_DEVSERVICES") == "1" else "sentry_kafka"
```

elsewhere in devserver startup to handle the container name differences. The new-devservices container also exposes `9092` instead of `9093`.

File: src/sentry/options/defaults.py [hunk 2]
```diff
@@ -411,6 +411,13 @@
     flags=FLAG_ALLOW_EMPTY | FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE,
 )
 
+# Taskbroker
+register(
+    "taskbroker.enabled",
```
Comment (markstory) [inline src/sentry/options/defaults.py:416]: This is a good solution for now. When we have taskbroker running in production, it would be good to tie into the `--workers` option for `sentry devserver` so that starting taskworkers is as simple as celery.

File: bin/send_to_taskbroker.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+
+task_namespace = taskregistry.create_namespace("test")
+
+no_retry_task = Task(
+    name="test.no_retry",
+    func=do_things,
+    namespace=task_namespace,
+    retry=None,
+)
```
Comment (markstory) [inline bin/send_to_taskbroker.py:22]: We could use the example tasks from `sentry.taskworker.tasks.examples` here as well.

**Review by markstory — APPROVED** [review]

File: src/sentry/options/defaults.py (hunk 2 above)
Comment (evanh) [reply src/sentry/options/defaults.py:416]: I'm going to leave this for now, since using --workers requires creating a new `sentry run ...` command.

File: bin/get_pending_task.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import logging
+import time
```
Comment (asottile-sentry) [inline bin/get_pending_task.py:2]: do we need to check these in? they look like debugging scripts

if needed -- please put them with the taskworker code so they don't go unowned / unused and please add appropriate tests

File: bin/get_pending_task.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+logging.info("Sending task")
+task_namespace.send_task(no_retry_task.create_activation())
+
+time.sleep(5)
+
+logging.info("Fetching task")
+taskworker = TaskWorker(rpc_host="localhost:50051", max_task_count=100)
+task = taskworker.fetch_task()
+
+logging.info(task)
```
Comment (asottile-sentry) [inline bin/get_pending_task.py:35]: modules should never have side-effects -- the execution parts should be inside of:

```python
def main() -> int:
    ...

if __name__ == '__main__': 
    raise SystemExit(main())
```

File: src/sentry/conf/server.py [hunk 6]
```diff
@@ -2421,6 +2421,24 @@ def custom_parameter_sort(parameter: dict) -> tuple[str, int]:
...
+                    else "sentry_kafka"
+                ),
+            },
+            "only_if": options.get("taskbroker.enabled"),
+            # we don't build linux/arm64 snuba images anymore
+            # apple silicon users should have working emulation under colima 0.6.2
+            # or docker desktop
+            "platform": "linux/amd64",
+        }
+    ),
```
Comment (asottile-sentry) [inline src/sentry/conf/server.py:2441]: can you add this to new devservices as well? cc @hubertdeng123 @IanWoodard 

this is the old one which is going away soon!

File: src/sentry/conf/server.py [hunk 7]
```diff
@@ -2421,6 +2421,24 @@ def custom_parameter_sort(parameter: dict) -> tuple[str, int]:
...
     ),
+    "taskbroker": lambda settings, options: (
+        {
+            "image": "ghcr.io/getsentry/taskbroker:latest",
+            "ports": {"50051/tcp": 50051},
+            "environment": {
+                "TASKBROKER_KAFKA_CLUSTER": (
+                    "kafka-kafka-1"
+                    if os.environ.get("USE_NEW_DEVSERVICES") == "1"
+                    else "sentry_kafka"
```
Comment (asottile-sentry) [inline src/sentry/conf/server.py:2432]: hmmm this is a little strange -- I don't expect these two worlds to overlap so I'm surprised we're bothering to adjust the host here

File: src/sentry/options/defaults.py [hunk 8]
```diff
@@ -411,6 +411,13 @@
...
+    default=False,
+    flags=FLAG_ALLOW_EMPTY | FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE,
+)
+
```
Comment (asottile-sentry) [inline src/sentry/options/defaults.py:420]: this seems unrelated to your change

File: bin/get_pending_task.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import logging
```
Comment (asottile-sentry) [inline bin/get_pending_task.py:1]: this seems unrelated to your change

**Review by asottile-sentry — CHANGES_REQUESTED** [review]
requesting changes to prevent auto merge

File: src/sentry/conf/server.py (hunk 6 above)
Comment (evanh) [reply src/sentry/conf/server.py:2441]: https://github.com/getsentry/sentry/pull/82244

File: src/sentry/options/defaults.py (hunk 8 above)
Comment (evanh) [reply src/sentry/options/defaults.py:420]: See check here: https://github.com/getsentry/sentry/pull/81968/files#diff-866fbc2da71a688b0a929236b2c864dfa8935d4a6423321d2d97a28337c6e6e8R2435

Comment (asottile-sentry) [reply src/sentry/options/defaults.py:420]: `options` don't work inside settings -- the database isn't available yet and so this will generate warning spam

everything else utilizes a settings boolean

Comment (asottile-sentry) [reply src/sentry/options/defaults.py:420]: hmm though I guess symbolicator uses this somehow.  strange I really don't expect that to work

File: src/sentry/conf/server.py [hunk 10]
```diff
@@ -2421,6 +2421,24 @@ def custom_parameter_sort(parameter: dict) -> tuple[str, int]:
...
+                ),
+            },
+            "only_if": options.get("taskbroker.enabled"),
+            # we don't build linux/arm64 snuba images anymore
```
Comment (asottile-sentry) [inline src/sentry/conf/server.py:2436]: copy paste error

**Review by asottile-sentry — APPROVED** [review]
